pkg/virt-controller/watch/export/export.go

```diff
@@ -0,0 +1,864 @@
+		Name: certificates,
+		VolumeSource: corev1.VolumeSource{
+			Secret: &corev1.SecretVolumeSource{
+				SecretName: secretName,
+			},
+		},
+	}, corev1.Volume{
+		Name: vmExport.Spec.TokenSecretRef,
+		VolumeSource: corev1.VolumeSource{
+			Secret: &corev1.SecretVolumeSource{
+				SecretName: vmExport.Spec.TokenSecretRef,
+			},
+		},
+	})
+
+	podManifest.Spec.Containers[0].VolumeMounts = append(podManifest.Spec.Containers[0].VolumeMounts, corev1.VolumeMount{
+		Name:      certificates,
+		MountPath: "/cert",
+	}, corev1.VolumeMount{
+		Name:      vmExport.Spec.TokenSecretRef,
+		MountPath: "/token",
+	})
+	return podManifest
+}
+
+func (ctrl *VMExportController) addVolumeEnvironmentVariables(exportContainer *corev1.Container, pvc *corev1.PersistentVolumeClaim, index int, mountPoint string) {
+	exportContainer.Env = append(exportContainer.Env, corev1.EnvVar{
+		Name:  fmt.Sprintf("VOLUME%d_EXPORT_PATH", index),
+		Value: mountPoint,
+	})
+	if pvc.Spec.VolumeMode != nil && *pvc.Spec.VolumeMode == corev1.PersistentVolumeBlock {
+		exportContainer.Env = append(exportContainer.Env, corev1.EnvVar{
+			Name:  fmt.Sprintf("VOLUME%d_EXPORT_RAW_URI", index),
+			Value: rawURI(pvc),
+		}, corev1.EnvVar{
+			Name:  fmt.Sprintf("VOLUME%d_EXPORT_RAW_GZIP_URI", index),
+			Value: rawGzipURI(pvc),
+		})
+	} else {
+		if ctrl.isKubevirtContentType(pvc) {
+			exportContainer.Env = append(exportContainer.Env, corev1.EnvVar{
+				Name:  fmt.Sprintf("VOLUME%d_EXPORT_RAW_URI", index),
+				Value: rawURI(pvc),
+			}, corev1.EnvVar{
+				Name:  fmt.Sprintf("VOLUME%d_EXPORT_RAW_GZIP_URI", index),
+				Value: rawGzipURI(pvc),
+			})
+		} else {
+			exportContainer.Env = append(exportContainer.Env, corev1.EnvVar{
+				Name:  fmt.Sprintf("VOLUME%d_EXPORT_ARCHIVE_URI", index),
+				Value: archiveURI(pvc),
+			}, corev1.EnvVar{
+				Name:  fmt.Sprintf("VOLUME%d_EXPORT_DIR_URI", index),
+				Value: dirURI(pvc),
+			})
+		}
+	}
+}
+
+func (ctrl *VMExportController) isKubevirtContentType(pvc *corev1.PersistentVolumeClaim) bool {
+	// Block volumes are assumed always KubevirtContentType
+	if pvc.Spec.VolumeMode != nil && *pvc.Spec.VolumeMode == corev1.PersistentVolumeBlock {
+		return true
+	}
+	isKubevirt := false
+	ann := pvc.GetAnnotations()
+	if ann != nil {
+		value, ok := ann[annContentType]
+		isKubevirt = ok && value == string(cdiv1.DataVolumeKubeVirt)
+	}
+	if !isKubevirt {
+		ownerRef := metav1.GetControllerOf(pvc)
+		if ownerRef != nil {
+			if ownerRef.Kind == datavolumeGVK.Kind && ownerRef.APIVersion == datavolumeGVK.GroupVersion().String() {
+				obj, exists, err := ctrl.DataVolumeInformer.GetStore().GetByKey(controller.NamespacedKey(pvc.GetNamespace(), ownerRef.Name))
+				if err != nil {
+					log.Log.V(1).Infof("Error getting DataVolume %v", err)
+				} else if exists {
+					dv, ok := obj.(*cdiv1.DataVolume)
+					isKubevirt = ok && (dv.Spec.ContentType == cdiv1.DataVolumeKubeVirt || dv.Spec.ContentType == "")
+				}
+			}
+		}
+	}
```

Comment:
style tip:

WIth Go, it is preferable to return early (or continue/break in a loop) rather than have nested ifs.  The goal being to keep code aligned as much as possible to the left.  For example

```golang
if isKubeVirt {
    return true
}

ownerRef := metav1.GetControllerOf(pvc)
if ownerRef == nil{
    return false
}

...

```

Reply:
Done